### PR TITLE
[GCS]profile info getting implementation based gcs service

### PR DESCRIPTION
**Pull request by WangTaoTheTonic**
## Why are these changes needed?

We have made gcs a service. All table info should be obtained from new gcs service instead of direct accessing of redis.

This PR changes the new way of accessing profile info aka. stats.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/gcs/gcs_server/stats_handler_impl.cc [hunk 1]
```diff
@@ -34,13 +34,33 @@ void DefaultStatsHandler::HandleAddProfileData(const AddProfileDataRequest &requ
...
     on_done(status);
   }
   RAY_LOG(DEBUG) << "Finished adding profile data, component type = "
                  << request.profile_data().component_type() << ", node id = " << node_id;
 }
 
+void DefaultStatsHandler::HandleGetAllProfileInfo(
+    const rpc::GetAllProfileInfoRequest &request, rpc::GetAllProfileInfoReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(INFO) << "Getting all profile info.";
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/stats_handler_impl.cc:49]: Can you change this to DEBUG? INFO is gonna be very verbose.

File: src/ray/gcs/gcs_server/stats_handler_impl.cc [hunk 2]
```diff
@@ -34,13 +34,33 @@ void DefaultStatsHandler::HandleAddProfileData(const AddProfileDataRequest &requ
...
+  auto on_done = [reply, send_reply_callback](
+                     const std::unordered_map<UniqueID, ProfileTableData> &result) {
+    for (auto &data : result) {
+      reply->add_profile_info_list()->CopyFrom(data.second);  // wangtao
+    }
+    RAY_LOG(INFO) << "Finished getting all profile info.";
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/stats_handler_impl.cc:55]: same here

File: src/ray/gcs/gcs_server/stats_handler_impl.cc [hunk 3]
```diff
@@ -34,13 +34,33 @@ void DefaultStatsHandler::HandleAddProfileData(const AddProfileDataRequest &requ
...
 }
 
+void DefaultStatsHandler::HandleGetAllProfileInfo(
+    const rpc::GetAllProfileInfoRequest &request, rpc::GetAllProfileInfoReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  RAY_LOG(INFO) << "Getting all profile info.";
+  auto on_done = [reply, send_reply_callback](
+                     const std::unordered_map<UniqueID, ProfileTableData> &result) {
+    for (auto &data : result) {
+      reply->add_profile_info_list()->CopyFrom(data.second);  // wangtao
```
Comment (raulchen) [inline src/ray/gcs/gcs_server/stats_handler_impl.cc:53]: This comment is unrelated?

File: src/ray/gcs/gcs_server/stats_handler_impl.cc (hunk 1 above)
Comment (WangTaoTheTonic) [reply src/ray/gcs/gcs_server/stats_handler_impl.cc]: OK

File: src/ray/gcs/gcs_server/stats_handler_impl.cc (hunk 2 above)
Comment (WangTaoTheTonic) [reply src/ray/gcs/gcs_server/stats_handler_impl.cc]: done

File: src/ray/gcs/gcs_server/stats_handler_impl.cc (hunk 3 above)
Comment (WangTaoTheTonic) [reply src/ray/gcs/gcs_server/stats_handler_impl.cc]: yeah...deleted...

**Review by raulchen — APPROVED** [review]
